### ISSUE TITLE
ios post build script require flutterfire to be available globally

### ISSUE THREAD

**Issue by cedvdb** [invertase/flutterfire_cli#257]
### Description

When running `flutterfire configure` (0.3.0-dev.19), post build scripts are added to ios. Those post build scripts require flutterfire to be installed on the machine building the ios application.

This is a problem because the flutterfire installed locally might not be the same version with which configure was run. Currently, for example, I'm using an hard coded version of the flutterfire cli in this github action so the build process can run https://github.com/cedvdb/action-flutter-build-ios/blob/main/action.yml#L54

I'd propose to use flutterfire_cli installed as a dev_dependency instead and fallback to a global one, so everyone, the CI and all developers can use the same version deterministically.

Related:

  - https://github.com/invertase/flutterfire_cli/issues/242
  - https://github.com/invertase/flutterfire_cli/issues/14
  
  

### Reasoning

It should be deterministic

**Comment by timmolter** [comment]
I noticed this issue today as well and came up with a manual workaround for building my iOS app via Codemagic. First, here's the error I got on codemagic:
```
    /Users/builder/Library/Developer/Xcode/DerivedData/Runner-edaimyiflreloheqntgnhkmwcclv/Build/Intermediates.noindex/ArchiveIntermediates/Runner/IntermediateBuildFilesPath/Runner.build/Release-iphoneos/Runner.build/Script-EAEA39BC9122492C49AD1220.sh: line 5: flutterfire: command not found
    Command PhaseScriptExecution failed with a nonzero exit code
```
This (**flutterfire: command not found**) makes sense because unlike my local dev environment, Codemagic never ran `dart pub global activate flutterfire_cli 0.3.0-dev.20 --overwrite` anywhere. 

My workround was to manually add `dart pub global activate flutterfire_cli 0.3.0-dev.20 --overwrite` to the `FlutterFire: flutterfire upload-crashlytics-symbols` script that `flutterfire configure` originally created.

<img width="910" alt="Screenshot 2024-03-15 at 22 33 04" src="https://github.com/invertase/flutterfire_cli/assets/738972/9c02a90d-4c9d-472f-b006-1da7099f1766">

**Comment by frederikstonge** [comment]
> I noticed this issue today as well and came up with a manual workaround for building my iOS app via Codemagic. First, here's the error I got on codemagic:
> 
> ```
>     /Users/builder/Library/Developer/Xcode/DerivedData/Runner-edaimyiflreloheqntgnhkmwcclv/Build/Intermediates.noindex/ArchiveIntermediates/Runner/IntermediateBuildFilesPath/Runner.build/Release-iphoneos/Runner.build/Script-EAEA39BC9122492C49AD1220.sh: line 5: flutterfire: command not found
>     Command PhaseScriptExecution failed with a nonzero exit code
> ```
> 
> This (**flutterfire: command not found**) makes sense because unlike my local dev environment, Codemagic never ran `dart pub global activate flutterfire_cli 0.3.0-dev.20 --overwrite` anywhere.
> 
> My workround was to manually add `dart pub global activate flutterfire_cli 0.3.0-dev.20 --overwrite` to the `FlutterFire: flutterfire upload-crashlytics-symbols` script that `flutterfire configure` originally created.
> 
> <img alt="Screenshot 2024-03-15 at 22 33 04" width="910" src="https://private-user-images.githubusercontent.com/738972/313340223-9c02a90d-4c9d-472f-b006-1da7099f1766.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1ODM2NjQsIm5iZiI6MTcxMjU4MzM2NCwicGF0aCI6Ii83Mzg5NzIvMzEzMzQwMjIzLTljMDJhOTBkLTRjOWQtNDcyZi1iMDA2LTFkYTcwOTlmMTc2Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwOFQxMzM2MDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMjZkYjViOTRiNmE0ZTExM2E3NGZiMmQ2OTY2NzdjYzMxMjAxMTkzYWVjMDRiNTIxYTQ4YzgwZDE0N2Q0NjNhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.JXosQQoC3VrAaUNv1QV83BiOp-ebuzPj-SoXVSWthdI">

Do you need to install firebase tools too? Or upload-crashlytics-symbols doesn't require them?

**Comment by benja8151** [comment]
> Do you need to install firebase tools too? Or upload-crashlytics-symbols doesn't require them?

No, flutterfire_cli is enough. I just added pre-build script (in Codemagic):
```shell
#!/bin/bash

dart pub global activate flutterfire_cli
```

**Comment by RodolfoSilva** [comment]
> > Do you need to install firebase tools too? Or upload-crashlytics-symbols doesn't require them?
> 
> No, flutterfire_cli is enough. I just added pre-build script (in Codemagic):
> 
> ```shell
> #!/bin/bash
> 
> dart pub global activate flutterfire_cli
> ```

Works!!!

**Comment by cedvdb** [comment]
not stale
